fixes for filecache

paulbiss opened the pull request:
The module argument was being improperly passed (should be relative to document root), and cmodule was added to specify the location of the static content. Drupal uses non-standard extensions for php files so ffile options are passed to bypass hhvm's usual exclusion rules for these files.

rrh commented on base/HHVMDaemon.php:
```diff
@@ -151,15 +150,18 @@ public function start(): void {
-          $files->add(ltrim(substr($path, strlen($sourceRoot)), '/'));
+        // Source files not ending in .php need to be specifically included
+        if (is_file($path) && substr($path, -4) !== '.php') {
+          $contents = file_get_contents($path);
```
file_get_contents supports a maxlen argument.  You only care about the first 4 bytes, so no point in reading the whole thing.

jwatzman commented:
> Drupal uses non-standard extensions for php files so ffile options are passed to bypass hhvm's usual exclusion rules for these files.

The change you made seems to affect repo auth mode, not just filecache. Do I need to make the same changes in my `hhvm-repo-auth` shell script I just landed? Also, it looks like you're doing this by appending a bunch of command-line arguments, won't we eventually hit the max command-line length on larger projects? Seems we should be able to use the file-list for this :(

paulbiss commented:
I think your shell script should work fine. I was misusing the module and cmodule options, and didn't actually need a file-list containing every file once I had fixed that.

I removed the file list because it was causing hhvm to treat every file as a source file rather than static content. We could probably use it rather than passing the `--ffile` options for php files not ending in `.php`. I decided to just copy our www makefile which does it this way.
